layouts/index.json.json

```diff
@@ -14,4 +14,4 @@
   {{- $list = $list | append $item -}}
 {{- end -}}
 
-{{- $list | jsonify -}}
+{{ jsonify (dict "indent" " ") $list | safeJS }}
```

Comment:
This `index.json` file is [fetched](https://github.com/zwbetz-gh/hugo-client-side-search-template/blob/67af96298fbf5557f237ab409b1f195c837c6777/assets/js/main.ts#L9) by your browser, so it's intentionally minified, aka no indent.